src/basic/efivars.c

```diff
@@ -365,10 +352,42 @@ int systemd_efi_options_variable(char **line) {
                 return -EPERM;
         }
 
-        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", line);
+        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", &line);
         if (r == -ENOENT)
                 return -ENODATA;
+        if (r < 0)
+                return r;
 
+        r = mkdir_safe_label("/run/systemd/efivars", 0700, 0, 0, MKDIR_WARN_MODE);
+        if (r < 0)
+                return r;
+
+        return write_string_file("/run/systemd/efivars/SystemdOptions", line,
```

Comment:
bikeshedding: if we end up caching other variables here the name here should probably contain the namespace uuid, i.e. incorporate EFI_VENDOR_SYSTEMD like /sys/firmware/efivars  would list it

Reply:
Done.